subprojects/reporting/src/main/java/org/gradle/api/reporting/internal/SimpleReport.java

```diff
@@ -53,11 +58,12 @@ public String toString() {
     }
 
     public File getDestination() {
-        return destination == null ? null : resolveToFile(destination);
+        Object evaluatedDestination = destination.isPresent() ? destination.get() : null;
+        return evaluatedDestination == null ? null : resolveToFile(evaluatedDestination);
     }
 
-    protected void setDestination(Object destination) {
-        this.destination = destination;
+    public void setDestination(Object destination) {
```

Comment:
should also add `setDestination(Provider<File>)` and `setDestination(File)` overloads and deprecate `setDestination(Object)`.